src/Str.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace OL\String;
+
+use OL\String\Contracts\Str as StrContract;
+
+/**
+ * Class Str.
+ *
+ * Basic manipulation of strings.
+ */
+class Str implements StrContract
+{
+    /**
+     * @var string Raw string content
+     */
+    protected $payload;
+
+    /**
+     * Str constructor.
+     *
+     * The constructor will enforce string type, since it can also receive
+     * another Str object (self) as it can be casted to string.
+     *
+     * @param string|StrContract $payload The actual string
+     */
+    public function __construct(string $payload)
+    {
+        // casts the payload as string and them assign on class scope
+        $this->payload = (string) $payload;
+    }
+
+    /**
+     * Str objects should should be able to be cast as string.
+     *
+     * @return string The actual string
+     */
+    public function __toString()
+    {
+        return $this->contents();
+    }
+
+    /**
+     * Public method to retrieve the raw string contents of an object.
+     *
+     * @return string
+     */
+    public function contents()
+    {
+        return $this->payload;
+    }
+
+    /**
+     * Replace one or more entries from the given object with a given value
+     *
+     * @param array|string $search What to search for
+     * @param string $replace Replace with?
+     *
+     * @return StrContract Object instance with the replaced text.
+     */
+    public function replace($search, string $replace)
```

Comment:
Also noting that `Immutable` here would have the same value that of `const` has in C/C++ parameters (readonly references that can't be changed). It would bring some value to the language, as primitive strings that travel through many different scopes in PHP are copied (and thus, local changes would not propagate elsewhere). It would be a "change prevention", that best emulates the passing-by-value of a primitive string -- but instead of discarding changes on scope change, it would prevent changes.